src/mysql-util/src/desc.rs

```diff
@@ -163,6 +167,33 @@ pub enum MySqlColumnMeta {
     Year,
 }
 
+impl IsCompatible for Option<MySqlColumnMeta> {
+    fn is_compatible(&self, other: &Option<MySqlColumnMeta>) -> bool {
+        match (self, other) {
+            (None, None) => true,
+            (Some(_), None) => false,
+            (None, Some(_)) => false,
+            (Some(self_meta), Some(other_meta)) => {
+                match (self_meta, other_meta) {
+                    (MySqlColumnMeta::Enum(self_enum), MySqlColumnMeta::Enum(other_enum)) => {
+                        // so as long as `self.values` is a compatible prefix of `other.values`, we can
+                        // ignore extra values from `other.values`.
+                        self_enum.values.len() <= other_enum.values.len()
+                            && self_enum
+                                .values
+                                .iter()
+                                .zip(other_enum.values.iter())
+                                .all(|(self_val, other_val)| self_val == other_val)
+                    }
+                    (MySqlColumnMeta::Json, MySqlColumnMeta::Json) => true,
+                    (MySqlColumnMeta::Year, MySqlColumnMeta::Year) => true,
+                    _ => false,
+                }
```

Comment:
```suggestion
            (None, None) => true,
            (Some(_), None) => false,
            (None, Some(_)) => false,
            (Some(MySqlColumnMeta::Enum(self_enum)), Some(MySqlColumnMeta::Enum(other_enum))) => {
                      // so as long as `self.values` is a compatible prefix of `other.values`, we can
                      // ignore extra values from `other.values`.
                      self_enum.values.len() <= other_enum.values.len()
                          && self_enum
                              .values
                              .iter()
                              .zip(other_enum.values.iter())
                              .all(|(self_val, other_val)| self_val == other_val)
                    }
               (Some(MySqlColumnMeta::Json), Some(MySqlColumnMeta::Json)) => true,     
               (Some(MySqlColumnMeta::Year), Some(MySqlColumnMeta::Year)) => true,     
                    _ => false,
                }
```

less nesting is probably clearer!

Reply:
ah yes great idea!